middleware/authMiddleware.js

```diff
@@ -0,0 +1,76 @@
+import User from "../../models/userModel.js";
+import bcrypt from "bcrypt";
+import Joi from "joi";
+import jwt from "jsonwebtoken";
+
+const registerUserSchema = Joi.object({
+  email: Joi.string().email().required(),
+  password: Joi.string().min(6).required(),
+});
+
+const checkTokenMiddleware = async (req, res, next) => {
+  const token = req.headers.authorization;
+
+  if (token) {
+    try {
+      const decodedToken = jwt.verify(token, "your-secret-key");
+      req.user = decodedToken;
+    } catch (error) {
+      console.error("Error verifying token:", error);
+      return res.status(401).json({ message: "Invalid token" });
+    }
+  }
+
+  next();
+};
+
+function verifyToken(req, res, next) {
```

Comment:
Zacznijmy od tej funkcji: 
1. Stwórz .env i tam zapisz twój secret zgodnie z sugestiami które dawałem przy zadaniu tym albo poprzednim. Zapisz w SECRET_KEY=jakissecretkey. Potem użyj go tutaj.
2. użyj decoded.id do pobrania z bazy danych takiego użytkownika. I sprawdź czy token który posiada ten użytkownik w bazie danych jest taki sam jak ten który został przysłany z frontendu. Jeśli użytkownik istnieje to przypisz go do req.user a jeśli nie istniej to zwróć błąd 'Not authorized'

Reply:
Jak zrobisz tę funkcję według sugestii to przyślij do sprawdzenia i pojdziemy dalej.